datalad/support/gitrepo.py

```diff
@@ -1262,6 +1262,21 @@ def _get_files_history(self, files, branch='HEAD'):
         """
         return gitpy.objects.commit.Commit.iter_items(self.repo, branch, paths=files)
 
+    def _gen_remotes_having_commit(self, commit_hexsha, with_urls_only=True):
+        """Traverse all branches of the remote and check if commit in any of their ancestry
+
+        It is a generator yielding names of the remotes
+        """
```

Comment:
Performance note:
Since here we call `git-config` anyway, it might be faster to use `git branch -r --contains <commit_hexsha>` instead.

Reply:
damn, I should have not been that lazy and indeed should have just called out to this one -- it is the ideal one for this task
Thanks @bpoldrack -- will do those changes now

Reply:
crap... implemented it, assuming that good git would guard us idiots from introducing '/' into remote names (would be needed to parse the output)... but not really:
```shell
$> git branch -r --contains eb055cda3068d2b22657a7e62252405f73b1120d
  bu/du                                                             
  bu/du2/du
  bu/du2/master
  bu/master
```
and since branch names could also have `/` in them -- there is no easy reliable way to split them into "remote" and "branch".  So may be I should stick to my ugly version for now :-/

Reply:
Hm. But does git allow for ambiguity? I mean: It just needs to match a remotes name right after "remotes/", no?

Reply:
it is a "branch" command, so it reports "re/mot/e/b/ran/ch" but it is not clear where the boundary between a remote and branch is ;)

Reply:
Yes, I got that. But we already know the names of remotes, so I thought it was an issue only, if there's actual ambiguity, meaning: There need to be two remote branches with the same name, consisting of three parts where the second part is part of the remote name for one of them and part of the actual branch name for the second one. I wondered, whether git would allow for this, since only then, we couldn't use that call.

So, I played:
```
/tmp/here % git remote -v
remote	../remote2 (fetch)
remote	../remote2 (push)
remote/withslash	../remote (fetch)
remote/withslash	../remote (push)

/tmp/remote % git branch
* master

/tmp/remote2 % git branch
  master
* withslash/master
```
Now, `remote/withslash/master` is ambigous.
What happens, if we fetch? Do we actually get ambiguous remote branches?

Nope:
```
tmp/here % git fetch --all
Fetching remote/withslash
Fetching remote
warning: no common commits
remote: Counting objects: 6, done.
remote: Compressing objects: 100% (3/3), done.
remote: Total 6 (delta 1), reused 0 (delta 0)
Unpacking objects: 100% (6/6), done.
From ../remote2
 * [new branch]      master           -> remote/master
 + bbb9828...e38f5a5 withslash/master -> remote/withslash/master  (forced update)
/tmp/here % git branch -r
  remote/master
  remote/withslash/master
```

So, this might be an issue in general, but not for our situation here. The mess happens before anyway and a different approach on finding that remote doesn't change anything as far as I can see.

Reply:
To be clear: The main point in discussing this issue is to figure it out, not to strongly oppose your approach.

Reply:
By looking it  up via GitPython, the same thing happens, of course:
```
/tmp/here % ipython
Python 2.7.9 (default, Jun 29 2016, 13:08:31) 
Type "copyright", "credits" or "license" for more information.

IPython 2.3.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: from git import Repo

In [2]: repo = Repo('.')

In [3]: repo.remotes
Out[3]: [<git.Remote "remote/withslash">, <git.Remote "remote">]

In [4]: [repo_obj.refs for repo_obj in repo.remotes]
Out[4]: 
[[<git.RemoteReference "refs/remotes/remote/withslash/master">],
 [<git.RemoteReference "refs/remotes/remote/master">,
  <git.RemoteReference "refs/remotes/remote/withslash/master">]]
In [5]: refs = [repo_obj.refs for repo_obj in repo.remotes]
In [6]: refs[0][0]
Out[6]: <git.RemoteReference "refs/remotes/remote/withslash/master">
In [7]: refs[0][0].commit
Out[7]: <git.Commit "e38f5a53fb6c98d16d971393c5aa2f0bd9553e92">
In [8]: refs[1][1]
Out[8]: <git.RemoteReference "refs/remotes/remote/withslash/master">
In [9]: refs[1][1].commit
Out[9]: <git.Commit "e38f5a53fb6c98d16d971393c5aa2f0bd9553e92">
```

So, the way you access it, that commit `e38f5a53fb6c98d16d971393c5aa2f0bd9553e92` is reported for both remotes.
But only one of them actually contains it:

```
/tmp/remote % git log
commit bbb98288ab3682f6acca10c724b067ee2d8d6fa1 (HEAD -> master)
Author: Benjamin Poldrack <benjaminpoldrack@gmail.com>
Date:   Fri Oct 20 08:52:29 2017 +0200

    first
/tmp/remote % cd ../remote2
/tmp/remote2 % git log
commit e38f5a53fb6c98d16d971393c5aa2f0bd9553e92 (HEAD -> withslash/master)
Author: Benjamin Poldrack <benjaminpoldrack@gmail.com>
Date:   Fri Oct 20 08:56:55 2017 +0200

    dsfg

commit ef1f057c40dcd5893068887a83fd43e83936b57d (master)
Author: Benjamin Poldrack <benjaminpoldrack@gmail.com>
Date:   Fri Oct 20 08:55:32 2017 +0200

    second
```